AirportTicketBookingSystem.Tests/DomainTests/ServicesTests/FlightServiceTests.cs

```diff
@@ -0,0 +1,132 @@
+using AirportTicketBookingSystem.Domain.Repositories;
+using AirportTicketBookingSystem.Domain.Services;
+public class FlightServiceTests
+{
+    private readonly Mock<IFlightRepository> _mockFlightRepo;
+    private readonly FlightService _flightService;
+    private readonly Fixture _fixture;
+
+    public FlightServiceTests()
+    {
+        _mockFlightRepo = new Mock<IFlightRepository>();
+        _flightService = new FlightService(_mockFlightRepo.Object);
+        _fixture = new Fixture();
+    }
+
+    [Fact]
+    public void AddFlight_ShouldCallRepositoryAddFlight()
+    {
+        var flight = _fixture.Create<Flight>();
+
+        _flightService.AddFlight(flight);
+
+        _mockFlightRepo.Verify(repo => repo.AddFlight(flight), Times.Once);
+    }
+
+    [Fact]
+    public void AddFlight_ShouldThrowException_WhenFlightAlreadyExists()
+    {
+        var flight = _fixture.Create<Flight>();
+        _mockFlightRepo.Setup(repo => repo.AddFlight(It.IsAny<Flight>()))
+            .Throws(new InvalidOperationException($"\nBooking {flight.Id} already exists"));
+
+        Action act = () => _flightService.AddFlight(flight);
+
+        act.Should().Throw<InvalidOperationException>()
+            .WithMessage($"\nBooking {flight.Id} already exists");
+    }
+
+    [Fact]
+    public void GetAllFlights_ShouldReturnAllFlights()
+    {
+        var flights = _fixture.CreateMany<Flight>(5).ToList();
+        _mockFlightRepo.Setup(repo => repo.GetAllFlights()).Returns(flights);
+
+        var result = _flightService.GetAllFlights();
+
+        result.Should().BeEquivalentTo(flights);
+    }
+
+     [Fact]
+    public void GetAllFlights_ShouldReturnEmptyList_WhenNoFlightsAvailable()
+    {
+        _mockFlightRepo.Setup(repo => repo.GetAllFlights()).Returns(new List<Flight>());
+
+        var result = _flightService.GetAllFlights();
+
+        result.Should().BeEmpty();
+    }
+
+    [Fact]
+    public void GetFlightById_ShouldReturnFlight_WhenFlightExists()
+    {
+        var flight = _fixture.Create<Flight>();
+        _mockFlightRepo.Setup(repo => repo.GetFlightById(flight.Id)).Returns(flight);
+
+        var result = _flightService.GetFlightById(flight.Id);
+
+        result.Should().BeEquivalentTo(flight);
+    }
+
+    [Fact]
+    public void GetFlightById_ShouldReturnNull_WhenFlightDoesNotExist()
+    {
+        int invalidId = _fixture.Create<int>();
+        _mockFlightRepo.Setup(repo => repo.GetFlightById(invalidId)).Returns((Flight?)null);
+
+        var result = _flightService.GetFlightById(invalidId);
+
+        result.Should().BeNull();
+    }
+
+    [Fact]
+    public void SearchFlights_ShouldReturnMatchingFlights()
+    {
+        var flights = _fixture.CreateMany<Flight>(10).ToList();
+        var targetFlight = flights.First();
+        targetFlight.DepartureCountry = "USA";
+        _mockFlightRepo.Setup(repo => repo.GetAllFlights()).Returns(flights);
+
+        var result = _flightService.SearchFlights(departureCountry: "USA");
```

Comment:
I noticed that SearchFlights method in the service is doing the filtration, technically speaking, we should be always doing the filtration on the DB side (repo method), we shouldn't be retrieving all data and then apply the filtration.

Lets move the filtration logic to the repo method `SearchFlights` instead , and move the tests related to the search  to the Flight Repo tests file.